### DISCUSSION TITLE
How to use gqlgen federation with RabbitMQ?

### DISCUSSION THREAD

**Discussion by dailytravel** [99designs/gqlgen#1801] [Q&A]
I'm writing a microservice project using federation, I want to build an independent service queue to perform the tasks of sending email to verify accounts, orders,...

Has anyone ever worked with RabbitMQ + Protobuf + gqlgen can not share for everyone's reference, thanks!

**Comment by sitompul** [comment]
use [executor](https://github.com/99designs/gqlgen/blob/master/graphql/executor/executor.go)

* consume message from rabbitmq
* decode the protobuf and parse it into ```RawParams```
* execute those ```RawParams``` using ```CreateOperationContext``` and ``` DispatchOperation```
